tavern/internal/c2/server.go

```diff
@@ -64,12 +65,36 @@ func (srv *Server) ClaimTasks(ctx context.Context, req *c2pb.ClaimTasksRequest)
 	if err != nil {
 		return nil, fmt.Errorf("failed to upsert host entity: %w", err)
 	}
+	// 2. check if beacon is new
+	resolvedbeacons, err := srv.graph.Beacon.Query().Where(beacon.IdentifierEQ(req.Beacon.Identifier)).Exist(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query beacon entity: %w", err)
+	}
+	var beaconnameaddr *string = nil
+	//3. if the beacon is new lets pick a name for it
+	if !resolvedbeacons {
+		candiatenames := []string{}
+		candiatenames = append(candiatenames, namegen.GetRandomName(namegen.Complexity(namegen.Simple)))
+		candiatenames = append(candiatenames, namegen.GetRandomName(namegen.Complexity(namegen.Moderate)))
+		candiatenames = append(candiatenames, namegen.GetRandomName(namegen.Complexity(namegen.Complex)))
+		collisions, err := srv.graph.Beacon.Query().Where(beacon.NameIn(candiatenames...)).All(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to query beacon entity: %w", err)
+		}
+		for _, canidate := range candiatenames {
```

Comment:
what behavior do we want if we get all collisions, i think current behavior is it regens it when ent builds the object using a complex name. i am worried attempting to regen it infinitely would cause problems with lots of agents (tbf it would be LOTS of agents) so we could do it. just worried about infinite loops

Reply:
this change will not be in upcoming commit as to hear feedback

Reply:
original request implimented